backend/neofs/util.go

```diff
@@ -18,14 +20,123 @@ import (
 	"github.com/rclone/rclone/fs"
 )
 
+type EndpointInfo struct {
+	Address  string
+	Priority int
+	Weight   float64
+}
+
+func parseEndpoints(endpointParam string) ([]EndpointInfo, error) {
+	expectedLength := -1 // to make sure all endpoints have the same format
+
+	endpoints := strings.Split(strings.TrimSpace(endpointParam), " ")
+	res := make([]EndpointInfo, 0, len(endpoints))
+	seen := make(map[string]struct{}, len(endpoints))
+
+	for _, endpoint := range endpoints {
+		endpointInfoSplit := strings.Split(endpoint, ",")
+		address := endpointInfoSplit[0]
+
+		if len(address) == 0 {
+			continue
+		}
+		if _, ok := seen[address]; ok {
+			return nil, fmt.Errorf("endpoint '%s' is already defined", address)
+		}
+		seen[address] = struct{}{}
+
+		endpointInfo := EndpointInfo{
+			Address:  address,
+			Priority: 1,
+			Weight:   1,
+		}
+
```

Comment:
Maybe we can move expected len check out of switch cases to avoid repeating? 
Consider something like this?

```go
		if expectedLength == -1 {
			expectedLength = len(endpointInfoSplit)
		}

		if len(endpointInfoSplit) != expectedLength {
			return nil, fmt.Errorf("all endpoints must have the same format: '%s'", endpointParam)
		}

		switch expectedLength {
		case 1:
		case 2:
			priority, err := parsePriority(endpointInfoSplit[1])
			// ...
		case 3:
			priority, err := parsePriority(endpointInfoSplit[1])
			// ...
		default:
			return nil, fmt.Errorf("invalid endpoint format '%s'", endpoint)
		}
```